core/execution/common/market_activity_tracker.go

```diff
@@ -530,6 +555,18 @@ func (mat *MarketActivityTracker) getAllParties(asset string, mkts []string) map
 	return parties
 }
 
+func (mat *MarketActivityTracker) getAllParties(asset string, mkts []string) map[string]struct{} {
+	return mat.filterParties(asset, mkts, func(mt *marketTracker) map[string]struct{} {
```

Comment:
why pass a function that returns a map rather than just the map?

Reply:
Because it's nested inside right?